### DISCUSSION TITLE
Performing one-off Renovate executions, without impacting existing Renovate PR/Dependency Dashboards

### DISCUSSION THREAD

**Discussion by jamietanna** [renovatebot/renovate#19352] [Request Help] [answered]
I'm currently looking at having projects opt-in for Renovate functionality across the organisation, where folks are able to decide if they want to use Renovate for their dependency management.

However, I also want to be able to perform arbitrary cross-organisation dependency upgrades. Using Renovate for this is ideal, due to the vast number of package managers and datasources available, and all the code being available to leverage.

For instance, let's say that in Go, there's a package called `github.com/<org>/grpc-models-go` and a Ruby gem called `grpc-models`, and in v2.0.0 there's a required update that all repositories must update to, for some arbitrary reason.

I'd like to be able to execute Renovate to perform this update across the whole organisation. 

I'm using a config such as, with v34.43.0:

```js
module.exports = {
  requireConfig: "ignored",
  onboarding: false,

  platform: "github",

  enabledManagers: ["ruby", "gomod"],

  packageRules: [
    // make sure that everything but what we want to raise updates for is disabled
    {
      "matchPackagePatterns": [".*"],
      "enabled": false
    },
    // for `grpc-models` the Ruby library, and github.com/<org>/grpc-models-go
    {
      "matchPackagePatterns": [".*grpc-models.*"],
      "enabled": true,
      "allowedVersions": ">=2.0.0"
    },
  ],
};
```

However, this ends up doing a few things that aren't expected:

- updating/closing the Dependency Dashboard, if one is created on that repo
- autoclosing any PRs that are not in the subset of packages listed above

Is there a way to avoid this? 

I'm happy writing a wrapper around the code, as I did with https://gitlab.com/tanna.dev/renovate-graph.

Digging into it:

- [`ensureDependencyDashboard`](https://github.com/renovatebot/renovate/blob/aab5fc38db2cf8db0e3185dbfdf29d4c5c4a0813/lib/workers/repository/dependency-dashboard.ts#L189-L197) is where we autoclose the Dependency Dashboard
- [`finaliseRepo`](https://github.com/renovatebot/renovate/blob/aab5fc38db2cf8db0e3185dbfdf29d4c5c4a0813/lib/workers/repository/finalise/index.ts#L21) is where `pruneStaleBranches` gets called to do the autoclosing of PRs and can be disabled through [`pruneStaleBranches`](https://docs.renovatebot.com/configuration-options/#prunestalebranches)

**Comment by jamietanna** [comment, marked as answer]
As noted in https://www.jvt.me/posts/2022/12/12/renovate-one-off/ I've created https://www.npmjs.com/package/@jamietanna/renovate-one-off as a lightweight wrapper over Renovate for this :raised_hands:

**Reply by viceice** [reply]
you should use the `force` config object to override repo settings from bot config. then you don't need the wrapper 😉

**Reply by jamietanna** [reply]
Interesting, would that allow for not closing / updating the dependency dashboard if set?
